echo_server.py

```diff
@@ -0,0 +1,89 @@
+import socket
+
+
+def create_server_socket():
+    server_socket = socket.socket(
+        socket.AF_INET,
+        socket.SOCK_STREAM,
+        socket.IPPROTO_IP)
+    # Set up the server socket
+    server_socket.bind(('127.0.0.1', 50000))
+    server_socket.listen(1)
+    return server_socket
+
+
+def response_ok():
+    return """\
+HTTP/1.1 200 OK\r\n\
+Content-Type: text/plain\r\n\r\n\
+Hey you.\r\n\
+"""
+
+
+def response_error(e):
+    return """\
+HTTP/1.1 {error}\r\n\
+\r\n\
+{cause}\r\n\
+    """.format(error=e.args[0], cause=e.args[1])
+
+
+def parse_request(request_string):
+    first_line = request_string.split('\r\n')[0]
+    first_line = first_line.split()
+
+    if request_protocol_validator(first_line):
+        return first_line[1]
+
+
+def request_protocol_validator(first_line):
+    try:
+        method, protocol = first_line[0], first_line[2]
+    except IndexError:
+        # By returning False, have the server continue to close the current 
+        # connection and wait for the next request without responding.
+        return False
+
+    if method != 'GET':
+        raise ValueError('403', 'GET method only.')
+    if protocol != 'HTTP/1.1':
+        raise ValueError('403', 'HTTP/1.1 only.')
+
+    if method == 'GET' and protocol == 'HTTP/1.1':
```

Comment:
This line is redundant.  If you get past 47 and 49, this is axiomatic.  Just return `True` without the extra test.